mmdet/datasets/custom.py

```diff
@@ -209,6 +217,40 @@ def prepare_train_img(self, idx):
         self.pre_pipeline(results)
         return self.pipeline(results)
 
+    def prepare_multiple_train_imgs(self, idx):
+        """Get training data and annotations (of multiple images) after
+        pipeline.
+
+        While the main sample is determined by idx, the other samples
+        are selected randomly.
+
+        Args:
+            idx (int): Index of the main sample.
+
+        Returns:
+            dict: Training data and annotation after pipeline with new keys \
+                introduced by pipeline.
+        """
+        num = self.num_samples_per_iter - 1
+
+        n_samples = len(self)
+        total_indices = list(range(n_samples))
+        total_indices.remove(idx)
+
+        selected_indices = random.choice(total_indices, num, False)
+        selected_indices = np.insert(selected_indices, 0, idx)
```

Comment:
I print all the indices. This may cause different workers select the same indices. You can set workers_per_gpu >= 2.  Then print all the selected indices to reproduce the bugs.